src/model/utils.rs

```diff
@@ -450,3 +450,64 @@ where
         })
         .collect()
 }
+
+pub fn optional_deserialize_components<'de, D>(
+    deserializer: D,
+) -> Result<Option<Vec<ActionRow>>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    let components = deserialize_components(deserializer)?;
+    if components.is_empty() {
+        Ok(None)
+    } else {
+        Ok(Some(components))
+    }
+}
```

Comment:
I tested a bit and seems like the gateway sends empty arrays as part of message update events. Right now they're preserved, but this would map them to `None`. I'd say this is probably unneeded, and the `MessageUpdateEvent::components` field can just be decorated with `#[serde(default, deserialize_with = "deserialize_components")]`.

Reply:
Oh right, this would not allow for the devs to know that the components were removed, would need to change the behaviour a little bit to match what is expected.

And would just using the deserialize_components one even work? I'm pretty sure it would complain because its trying to deserialize to Vec<T> and not Option<Vec<T>>

Reply:
If you add `#[serde(default)]` then it won't complain if the `components` field is missing.

Reply:
![image](https://github.com/user-attachments/assets/ea9b5044-3fe7-4eff-844a-2c2513dae7b9)

Reply:
I think the way forward would just be to change the function so that if components are null return null, and when empty do the empty collection and keep the extra function?

Might actually add another to handle this all correctly and keep duplication to a minimum? Sound correct?

Reply:
Yeah, I had it wrong. I guess a dedicated function will serve.

Reply:
Still need to distinguish between None and empty though... currently don't do that and I  can't really map it out in my head on how to do that without duplicating extra code

Reply:
Because Discord now returns the whole message payload on message edits, this is unnecessary and can always return Some, fef3b0aa39221f13221f20188a027bc59ca24395 does that now.